### DISCUSSION TITLE
How to initialize a new cache instance after one was already initialized during run time?

### DISCUSSION THREAD

**Discussion by mayalinetsky-kryon** [zilliztech/GPTCache#587] [Q&A] [answered]
I have a loop, and in each iteration I simulate conversations using a semantic cache, but each time with a different similarity threshold.
So one loop contains:
```
    cache_base = CacheBase("sqlite")
    vector_base = VectorBase("faiss", dimension=embedding_onnx.dimension)
    data_manager = get_data_manager(cache_base, vector_base, max_size=100000)
    cache = Cache()
    cache.init(
        embedding_func=embedding_onnx.to_embeddings,
        data_manager=data_manager,
        similarity_evaluation=SbertCrossencoderEvaluation(),
        config=Config(similarity_threshold=similarity_threshold),
    )

    cache.set_azure_openai_key()
```

How do I reset the cache for the second loop?

**Comment by SimFG** [comment]
When you use the cache api, you can try the `cache_factor` param.

If the current similarity threshold is 0.8, set cache_factor to 0.5, and the similarity threshold for this request will be 0.4

Take `openai` as an example,
```
openai.ChatCompletion.create(
      model='gpt-3.5-turbo',
      messages=[
        {
            'role': 'user',
            'content': question
        }
      ],
      cache_factor=0.8,
    )
```

**Reply by mayalinetsky-kryon** [reply]
Thank you for your reply, but this won't reset the cache the way that I want.
Setting cache factor to 0.8 will decrease the threshold, but will not remove any cached questions.

My intention is to initialize the cache from scratch in each loop, import some questions to cache and then asking other questions.
I should have mentioned that each loop differs not only by the threshold, but by the set of imported questions that are directly put to cache before starting the other requests.

Do you have a solution for this case?

**Reply by mayalinetsky-kryon** [reply, marked as answer]
In the end I manually manipulated the sqlite database using this function:
```
def reset_cache():
    sqlite_file = f"sqlite.db"

    if Path(sqlite_file).exists():

        conn = sqlite3.connect(sqlite_file)
        c = conn.cursor()

        get_table_query = """SELECT name FROM sqlite_master WHERE type='table';"""
        c.execute(get_table_query)
        table_names = [t[0] for t in c.fetchall()]
        for table in table_names:
            c.execute(f'DELETE FROM {table};')

        conn.commit()
        conn.close()

    faiss_file = f"faiss.index"
    if Path(faiss_file).exists():
        os.remove(faiss_file)
```

The question is - does this truly behave as though the cache was recreated from scratch?

**Reply by SimFG** [reply]
yes, it's a new cache
